File: src/harmonic-swimming.hs
-- harmonic swimming (jmcc)

import Sound.SC3.ID {- hsc3 -}
import Sound.SC3.UGen.External.RDU {- sc3-rdu -}

harmonic_swimming :: UGen
harmonic_swimming =
  let a = 0.02
      f = 50
      p = 20::Int
      l = line KR 0 (- a) 60 DoNothing
      o h = let r = randN 2 h 2 8
                n = lfNoise1 h KR r
                e = max 0 (n * a + l)
            in fSinOsc AR (f * (fromIntegral h + 1)) 0 * e
  in sum (map o [0..p])

main :: IO ()
main = audition (out 0 harmonic_swimming)

File: gr/harmonic-swimming.fs
\ harmonic swimming (jmcc) #1

: a 0.02 ;
: f 50 ;
: p 20 ;
: l 0 a negate 60 do-nothing Line.kr ;
: o { h }
    2 8 Rand.ir 2 clone LFNoise1.kr { n }
    0 n a * l + Max { e }
    f h 1 + * 0 FSinOsc.ar e * ;
: hs 0 p 0 do i o + loop ;
hs play
